cmd/collector/app/span_processor_test.go

```diff
@@ -807,3 +822,204 @@ func TestSpanProcessorWithOnDroppedSpanOption(t *testing.T) {
 	require.EqualError(t, err, processor.ErrBusy.Error())
 	assert.Equal(t, []string{"op3"}, droppedOperations)
 }
+
+func optionsWithPorts(portHttp string, portGrpc string) *cFlags.CollectorOptions {
+	opts := &cFlags.CollectorOptions{
+		OTLP: struct {
+			Enabled bool
+			GRPC    configgrpc.ServerConfig
+			HTTP    confighttp.ServerConfig
+		}{
+			Enabled: true,
+			HTTP: confighttp.ServerConfig{
+				Endpoint:        portHttp,
+				IncludeMetadata: true,
+			},
+			GRPC: configgrpc.ServerConfig{
+				NetAddr: confignet.AddrConfig{
+					Endpoint:  portGrpc,
+					Transport: confignet.TransportTypeTCP,
+				},
+			},
+		},
+	}
+	return opts
+}
+
+func TestOTLPReceiverWithV2Storage(t *testing.T) {
+	// Send trace via HTTP
+	t.Run("Send trace data using HTTP connection", func(t *testing.T) {
```

Comment:
there are multiple ways to organize the tests, yours makes the least sense to me.

Option 1: two top-level functions, with shared helper functions
Option 2: a single test function with table-driven tests, no setup helpers needed but each test has a lambda function parameter `executeRequest` implemented with two other functions, for http and grpc separately

Your test is only checking success path, does not check scenario when incoming tenant is invalid.